packages/webview_flutter/webview_flutter/lib/src/v4/src/navigation_delegate.dart

```diff
@@ -0,0 +1,111 @@
+///
+/// See [WebViewController.setNavigationDelegate].
+class NavigationDelegate {
+  /// Constructs a [NavigationDelegate].
+  NavigationDelegate({
+    FutureOr<bool> Function({
+      required String url,
+      required bool isForMainFrame,
+    })?
+        onNavigationRequest,
+    void Function(String url)? onPageStarted,
+    void Function(String url)? onPageFinished,
+    void Function(int progress)? onProgress,
+    void Function(WebResourceError error)? onWebResourceError,
+  }) : this.fromPlatformCreationParams(
+          const PlatformNavigationDelegateCreationParams(),
+          onNavigationRequest: onNavigationRequest,
+          onPageStarted: onPageStarted,
+          onPageFinished: onPageFinished,
+          onProgress: onProgress,
+          onWebResourceError: onWebResourceError,
+        );
+
+  /// Constructs a [NavigationDelegate] from creation params for a specific
+  /// platform.
+  NavigationDelegate.fromPlatformCreationParams(
+    PlatformNavigationDelegateCreationParams params, {
+    FutureOr<bool> Function({
+      required String url,
+      required bool isForMainFrame,
+    })?
+        onNavigationRequest,
+    void Function(String url)? onPageStarted,
+    void Function(String url)? onPageFinished,
+    void Function(int progress)? onProgress,
+    void Function(WebResourceError error)? onWebResourceError,
+  }) : this.fromPlatform(
+          PlatformNavigationDelegate(params),
+          onNavigationRequest: onNavigationRequest,
+          onPageStarted: onPageStarted,
+          onPageFinished: onPageFinished,
+          onProgress: onProgress,
+          onWebResourceError: onWebResourceError,
+        );
+
+  /// Constructs a [NavigationDelegate] from a specific platform implementation.
+  NavigationDelegate.fromPlatform(
+    this.platform, {
+    this.onNavigationRequest,
+    this.onPageStarted,
+    this.onPageFinished,
+    this.onProgress,
+    this.onWebResourceError,
+  }) {
+    if (onNavigationRequest != null) {
+      platform.setOnNavigationRequest(onNavigationRequest!);
+    }
+    if (onPageStarted != null) {
+      platform.setOnPageStarted(onPageStarted!);
+    }
+    if (onPageFinished != null) {
+      platform.setOnPageFinished(onPageFinished!);
+    }
+    if (onProgress != null) {
+      platform.setOnProgress(onProgress!);
+    }
+    if (onWebResourceError != null) {
+      platform.setOnWebResourceError(onWebResourceError!);
+    }
+  }
+
+  /// Implementation of [PlatformNavigationDelegate] for the current platform.
+  final PlatformNavigationDelegate platform;
+
+  /// Invoked when a decision for a navigation request is pending.
+  ///
+  /// When a navigation is initiated by the WebView (e.g when a user clicks a
+  /// link) this delegate is called and has to decide how to proceed with the
+  /// navigation.
+  ///
+  /// *Important*: Some platforms may also trigger this callback from calls to
+  /// [WebViewController.loadRequest].
+  final FutureOr<bool> Function({
```

Comment:
The docs should say what the return value means; currently it's non-obvious from the docs and name what `true` and `false` would do.

Reply:
The `NavigationDelegate` interface was updated, so the return value is now a `NavigationDecision.prevent` vs `NavigationDecision.navigate`.